### PR TITLE
libpanel: fix cross compilation, enable strictDeps

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 333563` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>87 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>budgie-analogue-clock-applet</li>
    <li>budgie-control-center</li>
    <li>budgie-control-center.debug</li>
    <li>budgie-desktop</li>
    <li>budgie-desktop-with-plugins</li>
    <li>budgie-desktop.dev</li>
    <li>budgie-desktop.man</li>
    <li>budgie-gsettings-overrides</li>
    <li>budgie-user-indicator-redux</li>
    <li>fm</li>
    <li>gnome-browser-connector</li>
    <li>gnome-builder</li>
    <li>gnome-builder.devdoc</li>
    <li>gnome-tweaks</li>
    <li>gnome.gnome-control-center</li>
    <li>gnome.gnome-control-center.debug</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-session.debug</li>
    <li>gnome.gnome-session.sessions</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-shell.debug</li>
    <li>gnome.gnome-shell.devdoc</li>
    <li>gnome.mutter</li>
    <li>gnome.mutter.debug</li>
    <li>gnome.mutter.dev</li>
    <li>gnome.mutter.devdoc</li>
    <li>gnome.mutter.man</li>
    <li>gnome.mutter43</li>
    <li>gnome.mutter43.debug</li>
    <li>gnome.mutter43.dev</li>
    <li>gnome.mutter43.devdoc</li>
    <li>gnome.mutter43.man</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>libpanel</li>
    <li>libpanel.dev</li>
    <li>libpanel.devdoc</li>
    <li>magpie</li>
    <li>magpie.debug</li>
    <li>magpie.dev</li>
    <li>magpie.devdoc</li>
    <li>matrix-gtk-theme</li>
    <li>mojave-gtk-theme</li>
    <li>monitor</li>
    <li>nightfox-gtk-theme</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-gsettings-schemas</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-shortcut-overlay</li>
    <li>pantheon.gala</li>
    <li>pantheon.switchboard-plug-a11y</li>
    <li>pantheon.switchboard-plug-bluetooth</li>
    <li>pantheon.switchboard-plug-mouse-touchpad</li>
    <li>pantheon.switchboard-plug-pantheon-shell</li>
    <li>pantheon.switchboard-plug-power</li>
    <li>pantheon.switchboard-plug-security-privacy</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-a11y</li>
    <li>pantheon.wingpanel-indicator-bluetooth</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-indicator-keyboard</li>
    <li>pantheon.wingpanel-indicator-network</li>
    <li>pantheon.wingpanel-indicator-nightlight</li>
    <li>pantheon.wingpanel-indicator-notifications</li>
    <li>pantheon.wingpanel-indicator-power</li>
    <li>pantheon.wingpanel-indicator-session</li>
    <li>pantheon.wingpanel-indicator-sound</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phoc</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>pods</li>
    <li>schemes</li>
    <li>sysprof</li>
    <li>sysprof.dev</li>
    <li>sysprof.lib</li>
    <li>tokyonight-gtk-theme</li>
    <li>vala-language-server</li>
    <li>vimix-gtk-themes</li>
    <li>whitesur-gtk-theme</li>
    <li>wingpanel-indicator-ayatana</li>
    <li>zrythm</li>
  </ul>
</details>

File: pkgs/development/libraries/libpanel/default.nix [hunk 1]
```diff
@@ -41,18 +52,23 @@ stdenv.mkDerivation rec {
   ];
 
   mesonFlags = [
-    "-Dinstall-examples=true"
+    (lib.mesonBool "install-examples" true)
+    (lib.mesonEnable "docs" true)
```
Comment (jtojnar) [inline pkgs/development/libraries/libpanel/default.nix:56]: This is already [`auto` by default](https://gitlab.gnome.org/GNOME/libpanel/-/blob/1.6.0/meson_options.txt#L16-18) and we have `auto_features=enabled`.

File: pkgs/development/libraries/libpanel/default.nix [hunk 2]
```diff
@@ -41,18 +52,23 @@ stdenv.mkDerivation rec {
...
   ];
 
+  postPatch = ''
+    # generate gir even when cross compiling
+    substituteInPlace meson.build \
+      --replace-fail "not meson.is_cross_build() or" ""
```
Comment (jtojnar) [inline pkgs/development/libraries/libpanel/default.nix:62]: Can you try to upstream something like https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/777?

Comment (NickCao) [reply pkgs/development/libraries/libpanel/default.nix:61]: Turns out it was already handling this case correctly, I misread the logic.

**Review by jtojnar — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**